dockworker.py

```diff
@@ -89,7 +106,17 @@ def create_notebook_server(self, base_path, container_name, container_config):
         #
         # Important piece here is the parametrized base_path to let the
         # underlying process know where the proxy is routing it.
-        rendered_command = container_config.command.format(base_path=base_path, port=port)
+        rendered_command = container_config.command.format(base_path=base_path, port=port,
+            ip=container_config.container_ip)
+
+        if container_config.host_network:
+            # Don't allow the notebook server to find another port: we won't be
+            # aware of it here and will wind up routing to the wrong server.
+            # Instead, let the server fail, the container die, and the cluster
+            # self-heal. Other possible approaches: find a free port here and
+            # assign it to the container (has a race condition), query the 
+            # running notebook config for its port (another race condition).
+            rendered_command += ' --NotebookApp.port_retries=0'
```

Comment:
Agreed. It's ugly for that reason too. 

Could be we do want to doc that the user has to set it as part of the command so that arbitrary, unknown ports aren't used. The command should try to use the specified arguments as given and fail if not possible (e.g., port taken). All the other methods I can think of do introduce a notebook-specific dependency.